packages/sdk/governance/src/proposals.ts

```diff
@@ -95,7 +112,14 @@ const isProxySetFunction = (tx: ProposalTransactionJSON) =>
  * @returns The JSON encoding of the proposal.
  */
 export const proposalToJSON = async (kit: ContractKit, proposal: Proposal) => {
-  const contractDetails = await obtainKitContractDetails(kit)
+  let addressOverride: AddressOverride = {}
+  proposal.forEach((tx) => {
```

Comment:
I don't think I understand why this needs to be done here? I see it's done on line 142. I'm just thinking of a situation where the proposal is something like:
1. Calls function on "TargetContract" whose registry address is 0xf00
2. Changes address of "TargetContract" in registry to 0xbad
which I think would be broken by this change?

Reply:
hmm

Reply:
fixed this by updating the registry before doing the abi lookup